src/test/java/BaseTest.java

```diff
@@ -1,11 +1,94 @@
 import io.github.bonigarcia.wdm.WebDriverManager;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.chrome.ChromeDriver;
 import org.openqa.selenium.chrome.ChromeOptions;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
 import org.testng.annotations.BeforeSuite;
 
+import java.time.Duration;
+
 public class BaseTest {
+    public WebDriver driver;
+    public String url = "https://qa.koel.app/";
 
     @BeforeSuite
     static void setupClass() {
         WebDriverManager.chromedriver().setup();
     }
+
+    @BeforeMethod
+    public void launchBrowser() {
+        ChromeOptions options = new ChromeOptions();
+        options.addArguments("--remote-allow-origins=*");
+
+        driver = new ChromeDriver(options);
+        driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(10));
+        driver.manage().window().maximize();
+    }
+
+    @AfterMethod
+    public void closeBrowser() {
+        driver.quit();
+    }
+
+    public void navigateToPage() {
+        driver.get(url);
+    }
+
+    public void provideEmail(String email) {
+        WebElement emailField = driver.findElement(By.cssSelector("[type='email']"));
+        emailField.click();
+        emailField.clear();
+        emailField.sendKeys(email);
+    }
+
+    public void providePassword(String password) {
+        WebElement passwordField = driver.findElement(By.cssSelector("[type='password']"));
+        passwordField.click();
+        passwordField.clear();
+        passwordField.sendKeys(password);
+    }
+
+    public void clickSubmit() {
+        WebElement submit = driver.findElement(By.cssSelector("button[type='submit']"));
+        submit.click();
+    }
+
+    public void searchSong(String name) throws InterruptedException {
+        WebElement searchField = driver.findElement(By.cssSelector("[type='search']"));
+        searchField.sendKeys(name);
+        Thread.sleep(2000);
+    }
+
+    public void clickViewAll() throws InterruptedException {
+        WebElement viewAllButton = driver.findElement(By.xpath("//button[@data-test='view-all-songs-btn']"));
+        viewAllButton.click();
+        Thread.sleep(2000);
+    }
+
+    public void clickFirstSong() throws InterruptedException {
+        WebElement firstSong = driver.findElement(By.xpath("//td[@class='title'][1]"));
```

Comment:
Please finish the homework!You got this problem, please correct it. 

![image](https://github.com/mlordlie/A49-QA-Automation-Framework/assets/104761482/061135e7-7e0c-4236-905a-3e70233ca31e)

HINT

![Test status_en](https://github.com/mlordlie/A49-QA-Automation-Framework/assets/104761482/7b42ba20-0c4d-429d-943f-efdea3475c6d)

For example

![Test error_en](https://github.com/mlordlie/A49-QA-Automation-Framework/assets/104761482/595fef20-4b4c-40b8-a3bb-8ea345f296ec)